lib/HLSL/DxilLegalizeSampleOffsetPass.cpp

```diff
@@ -182,6 +181,9 @@ void DxilLegalizeSampleOffsetPass::CollectIllegalOffsets(
   CollectIllegalOffsets(illegalOffsets, CurF, DXIL::OpCode::SampleGrad, hlslOP);
   CollectIllegalOffsets(illegalOffsets, CurF, DXIL::OpCode::SampleLevel,
                         hlslOP);
+  CollectIllegalOffsets(illegalOffsets, CurF, DXIL::OpCode::TextureGather, hlslOP);
+  CollectIllegalOffsets(illegalOffsets, CurF, DXIL::OpCode::TextureGatherCmp, hlslOP);
+  CollectIllegalOffsets(illegalOffsets, CurF, DXIL::OpCode::TextureLoad, hlslOP);
```

Comment:
nit: given that we are now expanding scope of this pass to also legalize offsets for texture gather and load ops, it might be more clearer to rename this pass as `DxilLegalizeOffsetsPass` or something else which more accurately reflects the expanded scope.

Reply:
Perhaps, but in my opinion, the name is ok, as it encompasses offsets to texture sample locations for textures, and not offsets for a variety of other things we might use the word "offset" for, even if it's for more than just the "Sample*()" methods.